10_predictive-modeling-review.qmd

```diff
@@ -0,0 +1,1045 @@
+
+knn_grid
+
+```
+
+Finally, we combine the recipe and model objects into a `workflow()`. 
+
+```{r}
+knn_workflow <-
+  workflow() |>
+  add_recipe(recipe = knn_recipe) |>
+  add_model(spec = knn_mod)
+
+```
+
+#### 5. Test and Choose the "Best" Model 
+
+`tune_grid()` fits the models to the repeated cross validation with differing hyperparameters and captures RMSE against each evaluation data set. 
+
+```{r}
+knn_res <-
+  knn_workflow |>
+  tune_grid(
+    resamples = data1_folds,
+    grid = knn_grid,
+    metrics = metric_set(rmse)
+  )
+
+```
+
+We can quickly extract information from the tuned results object with functions like `collect_metrics()`, `show_best()`, and `autoplot()`.
+
+```{r}
+knn_res |>
+  collect_metrics()
+
+knn_res |>
+  show_best()
+
+knn_res |>
+  autoplot()
+
+```
+
+#### 6. Optional: Expand the Model Fit
+
+Sometimes we will pick the best predictive model specification (feature and target engineering + algorithm + hyperparameters) and fit the model on all of the training data. 
+
+First, finalize the workflow with the "best" hyperparameters from the tuned results. 
+
+```{r}
+final_wf <- 
+  knn_workflow |> 
+  finalize_workflow(select_best(knn_res))
+
+```
+
+Second, use `last_fit()` to fit the model on all of the data. 
+
+```{r}
+final_fit <- 
+  final_wf |>
+  last_fit(data1_split) 
+
+```
+
+::: {.callout-note}
+
+`select_best()` takes a narrow view of model selection and picks the model with the lowest error rate. 
+
+It is important to consider other elements when picking a "best". Other considerations include parsimony, cost, and equity. 
+
+:::
+
+#### 7. Evaluate the Final Model
+
+The final fit object contains the out-of-sample error metric from the testing data. This metric is our best estimate of model performance on unseen data. 
+
+In this case, the testing data error is slightly higher than the training data error, which makes sense. 
+
+```{r}
+final_fit |>
+  collect_metrics()
+
+```
+
+#### 8. Optional: Implement the Final Model
+
+`final_fit` and `pedict()` can be used to apply the model to new, unseen data. 
+
+Only implement the model if it achieves the objectives of the final model. 
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+### Example: Regression Trees
+
+We'll next work with a subset of data from the `Chicago` data set from `library(tidymodels)`. 
+
+```{r}
+chicago_small <- Chicago |>
+  select(
+    ridership,
+    Clark_Lake,
+    Quincy_Wells,
+    Irving_Park,
+    Monroe,
+    Polk,
+    temp,
+    percip,
+    Bulls_Home,
+    Bears_Home,
+    WhiteSox_Home,
+    Cubs_Home,
+    date
+  ) |>
+  mutate(weekend = wday(date, label = TRUE) %in% c("Sat", "Sun")) |>
+  glimpse()
+
+```
+
+#### 0. Problem Formulation
+
+The variable `ridership` is the outcome of interest. It measures ridership at the Clark/Lake L station in Chicago. The variables `Clark_Lake`, `Quincy_Wells`, `Irving_Park`, `Monroe`, and `Polk` measure ridership at several stations *14 days before* the `ridership` variable. 
```

Comment:
Flag again here of "outcome of interest" language that I would suggest becoming "target variable" or something similar

Reply:
Changes.